src/simulation/atmos_grids.jl

```diff
@@ -0,0 +1,701 @@
+        comms_ctx,
+        float_type,
+    )
+end
+
+##############
+# SphereGrid #
+##############
+
+Base.@kwdef struct SphereGrid{
+    CS <: Spaces.ExtrudedFiniteDifferenceSpace,
+    FS <: Spaces.ExtrudedFiniteDifferenceSpace,
+    I <: Integer,
+    FT <: Real,
+    SR <: Meshes.StretchingRule,
+    T,
+} <: AbstractAtmosGrid
+    center_space::CS
+    face_space::FS
+
+    nh_poly::I
+
+    h_elem::I
+    radius::FT
+    z_elem::I
+    z_max::FT
+
+    z_stretch::SR
+
+    enable_bubble::Bool
+
+    topography::T
+end
+
+function Base.summary(io::IO, grid::SphereGrid)
+    println(io, "Grid type: $(nameof(typeof(grid)))")
+    println(io, "Number of vertical elements: $(grid.z_elem)")
+    println(io, "Height: $(grid.z_max) meters")
+    println(io, "Vertical grid stretching: $(nameof(typeof(grid.z_stretch)))")
+    # Add information about the stretching, if any
+    for field in fieldnames(typeof(grid.z_stretch))
+        println(io, "  with: $(field): $(getproperty(grid.z_stretch, field))")
+    end
+    println(io, "Radius: $(grid.radius) meters")
+    println(io, "Horizontal elements per edge: $(grid.h_elem)")
+    println(io, "  with: $(grid.nh_poly)-degree polynomials")
+    println(
+        io,
+        "  ",
+        grid.enable_bubble ? "with" : "without",
+        ": bubble correction",
+    )
+    if !isnothing(grid.topography)
+        println(io, "  with: $(grid.topography) topography")
+    end
+end
+
+"""
+function VerticallyStretchedSphereGrid(; nh_poly,
+                                         h_elem,
+                                         radius,
+                                         z_elem,
+                                         z_max,
+                                         dz_bottom,
+                                         dz_top,
+                                         topography = nothing,
+                                         topo_smoothing = false,
+                                         enable_bubble = false,
+                                         comms_ctx = ClimaComms.context(),
+                                         float_type = Float64)
+
+Construct an `SphereGrid` for a cubed sphere with columns with non-uniform
+resolution (as prescribed by
+`ClimaCore.Meshes.GeneralizedExponentialStretching`).
+
+Keyword arguments
+=================
+
+- `nh_poly`: Horizontal polynomial degree.
+- `radius`: Radius of the sphere (in meters).
+- `h_elem`: Number of spectral elements per edge.
+- `x_elem`: Number of spectral elements on the x direction.
+- `x_max`: Length of the box (in meters) (`x_min` is 0).
+- `y_elem`: Number of spectral elements on the y direction.
+- `y_max`: Depth of the box (in meters) (`y_min` is 0).
+- `z_elem`: Number of spectral elements on the vertical direction.
+- `dz_bottom`: Resolution at the lower end of the column (in meters).
+- `dz_top`: Resolution at the top end of the column (in meters).
+- `z_max`: Height of the column (in meters).
```

Comment:
How about we provide two options: 
1. a simple `z_grid` which takes in the column grid object,
2. pass any other arguments to the column grid constructor
 - If we just provide `z_elem` and `z_max`, it should be a uniform grid. 
 - if we provide the `dz_X` args, it should give a stretched grid